comparison/loadtest.go

```diff
@@ -114,6 +117,81 @@ type localCmd struct {
 	value []string
 }
 
+type bucketOps struct {
+	S3Client *s3.Client
+}
+
+// ListObjects lists the objects in a bucket.
+func (bucket bucketOps) ListObjects(bucketName string, ctx context.Context) ([]types.Object, error) {
+	result, err := bucket.S3Client.ListObjectsV2(ctx, &s3.ListObjectsV2Input{
+		Bucket: aws.String(bucketName),
+	})
+	var contents []types.Object
+	if err != nil {
+		return nil, fmt.Errorf("error listing objects in bucket %q: %v\n", bucketName, err)
+	} else {
+		contents = result.Contents
+	}
+
+	return contents, err
+}
+
+func (bucket bucketOps) DeleteObjects(bucketName string, objectKeys []string, ctx context.Context) error {
+	var objectIds []types.ObjectIdentifier
+	for _, key := range objectKeys {
+		objectIds = append(objectIds, types.ObjectIdentifier{Key: aws.String(key)})
+	}
+	_, err := bucket.S3Client.DeleteObjects(ctx, &s3.DeleteObjectsInput{
+		Bucket: aws.String(bucketName),
+		Delete: &types.Delete{Objects: objectIds},
+	})
+	if err != nil {
+		return fmt.Errorf("error deleting objects from bucket %q: %v\n", bucketName, err)
+	}
+	return err
+}
+
+func emptyBucket(bucket *bucketOps, bucketName string, ctx context.Context) error {
+	objects, err := bucket.ListObjects(bucketName, ctx)
+	if err != nil {
+		return err
+	}
+
+	var objKeys []string
+	for _, object := range objects {
+		objKeys = append(objKeys, *object.Key)
+	}
+
+	err = bucket.DeleteObjects(bucketName, objKeys, ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func populateBucket(bucket *bucketOps, targetBucket, sourceBucket string, ctx context.Context) error {
+	objects, err := bucket.ListObjects(sourceBucket, ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get source bucket(%q) objects during populate bucket: %v", sourceBucket, err)
+	}
+
+	var copyErrors []error
+	for _, object := range objects {
+		_, err := bucket.S3Client.CopyObject(ctx, &s3.CopyObjectInput{Bucket: aws.String(targetBucket), CopySource: aws.String(sourceBucket), Key: object.Key})
+		if err != nil {
```

Comment:
@agnivade what do you suggest here? we need to copy the objects 1 by 1.

Reply:
if we have an error with a single file should we return it. If a single file is not copied would the load test work fine?

Reply:
It would not be a big deal, but loading a post which has that file attachment will fail. Which might increase the error rates. I'll let Alejandro take a deeper look and comment.